File: Leetcode/217. Contains Duplicate/217_Contains_Duplicate.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        // Create a HashSet to store unique integers from the 'nums' array
        Set<Integer> set = new HashSet<>();

        // Iterate through each element 'nums[i]' in the 'nums' array
        for (int i = 0; i < nums.length; i++) {
            // Check if 'set' already contains the current element 'nums[i]'
            if (set.contains(nums[i])) {
                // If a duplicate is found, return true
                return true;
            }
            // Add the current element 'nums[i]' to the 'set' to track unique values
            set.add(nums[i]);
        }

        // If no duplicates are found after iterating through the entire array, return
        // false
        return false;

    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        int[] nums1 = { 1, 2, 3, 4, 5 };
        System.out.println("Array 1 contains duplicate: " + solution.containsDuplicate(nums1));
        // Should print "Array 1 contains duplicate: false"

        int[] nums2 = { 1, 2, 3, 1, 4, 5 };
        System.out.println("Array 2 contains duplicate: " + solution.containsDuplicate(nums2));
        // Should print "Array 2 contains duplicate: true"

        int[] nums3 = { 1, 2, 3, 4, 5, 6, 7, 8, 9 };
        System.out.println("Array 3 contains duplicate: " + solution.containsDuplicate(nums3));
        // Should print "Array 3 contains duplicate: false"
    }
}

File: Leetcode/217. Contains Duplicate/217_Contains_Duplicate.py
import collections


class Solution(object):
    def containsDuplicate(self, nums):
        # Create an empty set to store unique elements
        hashset = set()
        # Iterate through each element 'n' in the 'nums' list
        for n in nums:
            # Check if 'n' is already in the 'hashset' (duplicate found)
            if n in hashset:
                return True  # Return True if a duplicate is found
            # If 'n' is not in the 'hashset', add it to the 'hashset'
            hashset.add(n)
        # If no duplicates are found after looping through the entire list, return False
        return False


class Main:
    def execute(self):
        result = Solution().containsDuplicate([1, 2, 3, 1])
        print(result)


if __name__ == "__main__":
    Main().execute()
